ingest/importer/importer.py

```diff
@@ -188,23 +203,18 @@ def do_import(self, workbook: IngestWorkbook, project_uuid=None):
         workbook_errors = []
         for worksheet in importable_worksheets:
             try:
+                self.sheet_in_schemas(worksheet)
                 metadata_entities, worksheet_errors = self.worksheet_importer.do_import(worksheet)
                 module_field_name = worksheet.get_module_field_name()
-                for error in worksheet_errors:
-                    if error["location"]:
-                        error["location"] = f'sheet={worksheet.title}, {error["location"]}'
-                    else:
-                        error["location"] = f'sheet={worksheet.title}'
-                    workbook_errors.append(error)
+                workbook_errors.extend(worksheet_errors)
 
-                for entity in metadata_entities:
-                    if worksheet.is_module_tab():
-                        entity.retain_content_fields(module_field_name)
-                        registry.add_module(entity)
-                    else:
-                        registry.add_submittable(entity)
+                if worksheet.is_module_tab():
+                    removed_data = registry.add_modules(module_field_name, metadata_entities)
+                    workbook_errors.extend(self.list_data_removal_errors(worksheet.title, removed_data))
```

Comment:
Any errors aren't raised as exceptions because they are added to the error list to be submitted to the submission envelope. Any errors associated with the submission envelope will stop the submission from being exported.